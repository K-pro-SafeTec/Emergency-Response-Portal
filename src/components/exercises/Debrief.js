import React  from 'react';

export default ({match}) => (
  <div>
    <h2>Debrief report for scenario {match.params.scenarioId} on date {match.params.date}.</h2>
  </div>
)
