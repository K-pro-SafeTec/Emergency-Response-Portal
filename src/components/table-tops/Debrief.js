import React from 'react';
import {Link} from "react-router-dom";
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import Button from "@material-ui/core/Button/Button";
import AppPage from '../shared/AppPage';
import EntityInfo from '../shared/EntityInfo';
import IconTableTop from '../../icons/main-menu/tableTops.svg';
import { tableTopScenarioById } from '../../dummy-data/tableTopScenario';
import { getTableTopInstanceByKey } from '../../dummy-data/tableTopInstance';
import TextField from '@material-ui/core/TextField';


const Debrief = ({ match }) => {
  const instance = getTableTopInstanceByKey(match.params.scenarioId, match.params.date);
  if (instance) {
    const scenario = tableTopScenarioById[match.params.scenarioId];
    return (
      <AppPage title="Debrief">
        <Paper style={{margin: '1em', padding: '1em'}}>
          <EntityInfo Icon={(props) => <img src={IconTableTop} alt="Table Top" {...props} />}>
            <Typography variant="headline">DFU {scenario.dfuNr}</Typography>
            <Typography variant="subheading">{scenario.dfuTitle}</Typography>
          </EntityInfo>
          {scenario.debrief.length > 0 && (
            <React.Fragment>
              <Typography variant="title">Oppsummering:</Typography>
              {scenario.debrief.map((debrief, key) => (
                <TextField key={key} label={debrief} margin="normal" multiline fullWidth />
              ))}
            </React.Fragment>
          )}
          <Button variant="contained" color="primary" component={Link} to="..">Fullf√∏r</Button>
        </Paper>
      </AppPage>
    );
  }
  return (
    <AppPage title="Table tops" back="..">
      <Typography>Fant ikke table top</Typography>
    </AppPage>
  )
};

export default Debrief;
