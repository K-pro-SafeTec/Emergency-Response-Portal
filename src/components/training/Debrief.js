import React  from 'react';
import PlaceholderPage from '../shared/PlaceholderPage';

export default ({ match }) => (
  <PlaceholderPage title="Rapport">
    {/* <h2>Debrief report for scenario {match.params.scenarioId} on date {match.params.date}.</h2> */}
  </PlaceholderPage>
)
