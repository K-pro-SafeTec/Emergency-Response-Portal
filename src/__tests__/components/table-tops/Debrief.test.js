import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom'
import Debrief from "../../../components/table-tops/Debrief";

let testRenderer;

beforeAll(() => {
  testRenderer = TestRenderer.create(
    <Router>
      <Debrief
        match={{
          params: {
            scenarioId: 4,
            date: '2018-09-24',
          }
        }}
      />
    </Router>
  );
});

test('renders correctly', () => {
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});
