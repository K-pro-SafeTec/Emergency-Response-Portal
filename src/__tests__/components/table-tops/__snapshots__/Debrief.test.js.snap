// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-5 MuiAppBar-positionFixed-6 MuiAppBar-colorDefault-11 mui-fixed"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <a
        className="MuiButtonBase-root-51 MuiIconButton-root-45 AppPage-homeButton-2"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-178"
        />
      </a>
      <button
        className="MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiIconButton-root-45 MuiIconButton-disabled-49"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
      <button
        className="MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiIconButton-root-45 MuiIconButton-disabled-49"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
      </button>
      <h2
        className="MuiTypography-root-63 MuiTypography-title-69 MuiTypography-colorInherit-92 AppPage-grow-1"
      >
        Debrief
      </h2>
      <a
        className="AppPage-safetecLink-3"
        href="https://www.safetec.no/"
      >
        <img
          alt="Levert av Safetec"
          height="44px"
          src="safetec.svg"
          width="153px"
        />
      </a>
    </div>
  </header>,
  <div
    className="AppPage-content-4"
  >
    <div
      className="MuiPaper-root-14 MuiPaper-elevation2-18 MuiPaper-rounded-15"
      style={
        Object {
          "margin": "1em",
          "padding": "1em",
        }
      }
    >
      <div
        className="EntityInfo-container-98"
      >
        <img
          alt="Table Top"
          className="EntityInfo-avatar-99"
          src="tableTops.svg"
        />
        <div
          className="EntityInfo-info-100"
        >
          <h1
            className="MuiTypography-root-63 MuiTypography-headline-68"
          >
            DFU 
            4
          </h1>
          <h3
            className="MuiTypography-root-63 MuiTypography-subheading-70"
          >
            Fare for kollisjon
          </h3>
        </div>
      </div>
      <h2
        className="MuiTypography-root-63 MuiTypography-title-69"
      >
        Oppsummering:
      </h2>
      <div
        className="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
      >
        <label
          className="MuiFormLabel-root-112 MuiInputLabel-root-105 MuiInputLabel-formControl-106 MuiInputLabel-animated-109"
          data-shrink={false}
        >
          Gjennomgang av det vi s√• fungerte bra
        </label>
        <div
          className="MuiInputBase-root-132 MuiInput-root-119 MuiInput-underline-123 MuiInputBase-fullWidth-141 MuiInput-fullWidth-126 MuiInputBase-formControl-133 MuiInput-formControl-120 MuiInputBase-multiline-140 MuiInput-multiline-125"
          onClick={[Function]}
        >
          <div
            className="MuiPrivateTextarea-root-149"
          >
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows={1}
              tabIndex={-1}
            />
            <textarea
              aria-invalid={false}
              className="MuiPrivateTextarea-textarea-150 MuiInputBase-input-142 MuiInput-input-127 MuiInputBase-inputMultiline-144 MuiInput-inputMultiline-129"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={1}
              style={
                Object {
                  "height": 19,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
      >
        <label
          className="MuiFormLabel-root-112 MuiInputLabel-root-105 MuiInputLabel-formControl-106 MuiInputLabel-animated-109"
          data-shrink={false}
        >
          Gjennomgang av det som kan bli bedre
        </label>
        <div
          className="MuiInputBase-root-132 MuiInput-root-119 MuiInput-underline-123 MuiInputBase-fullWidth-141 MuiInput-fullWidth-126 MuiInputBase-formControl-133 MuiInput-formControl-120 MuiInputBase-multiline-140 MuiInput-multiline-125"
          onClick={[Function]}
        >
          <div
            className="MuiPrivateTextarea-root-149"
          >
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows={1}
              tabIndex={-1}
            />
            <textarea
              aria-invalid={false}
              className="MuiPrivateTextarea-textarea-150 MuiInputBase-input-142 MuiInput-input-127 MuiInputBase-inputMultiline-144 MuiInput-inputMultiline-129"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={1}
              style={
                Object {
                  "height": 19,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
      >
        <label
          className="MuiFormLabel-root-112 MuiInputLabel-root-105 MuiInputLabel-formControl-106 MuiInputLabel-animated-109"
          data-shrink={false}
        >
          Gjennomgang av det som vi ikke fikk til
        </label>
        <div
          className="MuiInputBase-root-132 MuiInput-root-119 MuiInput-underline-123 MuiInputBase-fullWidth-141 MuiInput-fullWidth-126 MuiInputBase-formControl-133 MuiInput-formControl-120 MuiInputBase-multiline-140 MuiInput-multiline-125"
          onClick={[Function]}
        >
          <div
            className="MuiPrivateTextarea-root-149"
          >
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows={1}
              tabIndex={-1}
            />
            <textarea
              aria-invalid={false}
              className="MuiPrivateTextarea-textarea-150 MuiInputBase-input-142 MuiInput-input-127 MuiInputBase-inputMultiline-144 MuiInput-inputMultiline-129"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={1}
              style={
                Object {
                  "height": 19,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
      >
        <label
          className="MuiFormLabel-root-112 MuiInputLabel-root-105 MuiInputLabel-formControl-106 MuiInputLabel-animated-109"
          data-shrink={false}
        >
          Hvilke aksjoner/l√¶repunkter skal tas videre (oppdatering av planer, sjekkliste, varslingsrutiner etc.
        </label>
        <div
          className="MuiInputBase-root-132 MuiInput-root-119 MuiInput-underline-123 MuiInputBase-fullWidth-141 MuiInput-fullWidth-126 MuiInputBase-formControl-133 MuiInput-formControl-120 MuiInputBase-multiline-140 MuiInput-multiline-125"
          onClick={[Function]}
        >
          <div
            className="MuiPrivateTextarea-root-149"
          >
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows={1}
              tabIndex={-1}
            />
            <textarea
              aria-invalid={false}
              className="MuiPrivateTextarea-textarea-150 MuiInputBase-input-142 MuiInput-input-127 MuiInputBase-inputMultiline-144 MuiInput-inputMultiline-129"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={1}
              style={
                Object {
                  "height": 19,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
      >
        <label
          className="MuiFormLabel-root-112 MuiInputLabel-root-105 MuiInputLabel-formControl-106 MuiInputLabel-animated-109"
          data-shrink={false}
        >
          Hva b√∏r vi trene mer p√•
        </label>
        <div
          className="MuiInputBase-root-132 MuiInput-root-119 MuiInput-underline-123 MuiInputBase-fullWidth-141 MuiInput-fullWidth-126 MuiInputBase-formControl-133 MuiInput-formControl-120 MuiInputBase-multiline-140 MuiInput-multiline-125"
          onClick={[Function]}
        >
          <div
            className="MuiPrivateTextarea-root-149"
          >
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows={1}
              tabIndex={-1}
            />
            <textarea
              aria-invalid={false}
              className="MuiPrivateTextarea-textarea-150 MuiInputBase-input-142 MuiInput-input-127 MuiInputBase-inputMultiline-144 MuiInput-inputMultiline-129"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={1}
              style={
                Object {
                  "height": 19,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
      >
        <label
          className="MuiFormLabel-root-112 MuiInputLabel-root-105 MuiInputLabel-formControl-106 MuiInputLabel-animated-109"
          data-shrink={false}
        >
          Erfaringer vi skal dele med andre som ikke var med p√• √∏velsen
        </label>
        <div
          className="MuiInputBase-root-132 MuiInput-root-119 MuiInput-underline-123 MuiInputBase-fullWidth-141 MuiInput-fullWidth-126 MuiInputBase-formControl-133 MuiInput-formControl-120 MuiInputBase-multiline-140 MuiInput-multiline-125"
          onClick={[Function]}
        >
          <div
            className="MuiPrivateTextarea-root-149"
          >
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-150 MuiPrivateTextarea-shadow-151"
              readOnly={true}
              rows={1}
              tabIndex={-1}
            />
            <textarea
              aria-invalid={false}
              className="MuiPrivateTextarea-textarea-150 MuiInputBase-input-142 MuiInput-input-127 MuiInputBase-inputMultiline-144 MuiInput-inputMultiline-129"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={1}
              style={
                Object {
                  "height": 19,
                }
              }
            />
          </div>
        </div>
      </div>
      <a
        className="MuiButtonBase-root-51 MuiButton-root-152 MuiButton-contained-163 MuiButton-containedPrimary-164 MuiButton-raised-166 MuiButton-raisedPrimary-167"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-153"
        >
          Fullf√∏r
        </span>
        <span
          className="MuiTouchRipple-root-178"
        />
      </a>
    </div>
  </div>,
]
`;
